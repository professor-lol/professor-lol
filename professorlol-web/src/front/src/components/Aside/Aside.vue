<template>
  <aside class="aside-menu">
    <slot>Aside</slot>
  </aside>
</template>

<script>
export default {
  name: 'AppAside',
  props: {
    fixed: {
      type: Boolean,
      default: false
    },
    display: {
      type: String,
      default: ''
    },
    offCanvas: {
      type: Boolean,
      default: true
    }
  },
  mounted: function () {
    this.isFixed(this.fixed)
    this.isOffCanvas(this.offCanvas)
  },
  methods: {
    isFixed (fixed) {
      fixed ? document.body.classList.add('aside-menu-fixed') : document.body.classList.remove('aside-menu-fixed')
      return fixed
    },
    isOffCanvas (offCanvas) {
      offCanvas ? document.body.classList.add('aside-menu-off-canvas') : document.body.classList.remove('aside-menu-off-canvas')
    }
  }
}
</script>
