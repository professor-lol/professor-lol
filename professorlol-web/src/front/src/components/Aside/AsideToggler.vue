<template>
  <button
    :display="display"
    :mobile="mobile"
    :class="classList"
    type="button"
    @click="asideToggle">
    <span class="navbar-toggler-icon" />
  </button>
</template>

<script>
import { asideMenuCssClasses, validBreakpoints, checkBreakpoint } from '../../shared/classes'
import toggleClasses from '../../shared/toggle-classes'

export default {
  name: 'AsideToggler',
  props: {
    defaultOpen: {
      type: Boolean,
      default: false
    },
    display: {
      type: String,
      default: 'lg'
    },
    mobile: {
      type: Boolean,
      default: false
    },
  },
  computed: {
    classList () {
      return [
        'navbar-toggler'
      ]
    }
  },
  mounted: function() {
    this.toggle(this.defaultOpen)
  },
  methods: {
    toggle (force) {
      const [display, mobile] = [this.display, this.mobile]
      let cssClass = asideMenuCssClasses[0]
      if (!mobile && display && checkBreakpoint(display, validBreakpoints)) {
        cssClass = `aside-menu-${display}-show`
      }
      toggleClasses(cssClass, asideMenuCssClasses, force)
    },
    asideToggle (e) {
      e.preventDefault()
      this.toggle()
    }
  }
}
</script>
