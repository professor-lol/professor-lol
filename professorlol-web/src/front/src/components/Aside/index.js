import Aside from './Aside'
import AsideToggler from './AsideToggler'

export {
  Aside,
  AsideToggler
}
