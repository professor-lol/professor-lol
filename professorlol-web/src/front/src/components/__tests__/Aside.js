import { mount } from '@vue/test-utils'
import Aside from "../Aside/Aside";

describe("Aside.vue", () => {
  // Inspect the raw component options
  it('has isFixed method', () => {
    expect(typeof Aside.methods.isFixed).toBe('function')
  })
  it('has isOffCanvas method', () => {
    expect(typeof Aside.methods.isOffCanvas).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(Aside)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.is('aside')).toBe(true)
    expect(wrapper.classes()).toContain('aside-menu')
    expect(wrapper.element.textContent).toEqual('Aside')
  })
});
