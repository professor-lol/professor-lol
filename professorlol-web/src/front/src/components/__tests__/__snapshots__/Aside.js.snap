// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside.vue renders correctly 1`] = `
<aside
  class="aside-menu"
>
  Aside
</aside>
`;
